### PR TITLE
Certparty

### PR DESCRIPTION
**Pull request by shoegaze**
Cleaning up cert issues.

### CODE REVIEW

File: ztools/x509/json.go [hunk 1]
```diff
@@ -184,6 +184,7 @@ type jsonCertificate struct {
 	Issuer             pkix.Name                    `json:"issuer"`
 	IssuerDN           string                       `json:"issuer_dn,omitempty"`
 	Validity           validity                     `json:"validity"`
+	ValidFor           time.Duration                `json:"valid_for"`
```
Comment (zakird) [inline ztools/x509/json.go]: This should probably go inside of validity and be called length.

**Comment by zakird** [comment]
We also need to update zgrab_schema.py

File: ztools/x509/json.go (hunk 1 above)
Comment (shoegaze) [inline ztools/x509/json.go]: Yeah, that just occurred to me as I walked out of the room haha. I'll fix it.

File: ztools/x509/testdata/example.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
 		},
 		"validity": {
 			"start": "20140102",
-			"end": "20150102"
+			"end": "20150102",
+            "length" :8760
```
Comment (zakird) [inline ztools/x509/testdata/example.json]: This looks like a formatting error.

File: ztools/x509/json.go [hunk 3]
```diff
@@ -136,14 +136,16 @@ func (p *PublicKeyAlgorithm) UnmarshalJSON(b []byte) error {
 }
 
 type auxValidity struct {
-	Start string `json:"start"`
-	End   string `json:"end"`
+	Start          string `json:"start"`
+	End            string `json:"end"`
+	ValidityPeriod int    `json:"length"`
```
Comment (dadrian) [inline ztools/x509/json.go]: I would call this "hours" or something that implies it's in hours and not seconds.

File: ztools/x509/json.go [hunk 4]
```diff
@@ -177,13 +180,18 @@ type jsonSignature struct {
 	SelfSigned         bool               `json:"self_signed"`
 }
 
+type fullValidity struct {
+	validity
+	ValidityPeriod int
```
Comment (dadrian) [inline ztools/x509/json.go]: Why not just add `ValidityPeriod` to to `validity`?

File: ztools/x509/json.go (hunk 3 above)
Comment (zakird) [inline ztools/x509/json.go]: We could also switch it to be seconds.

File: ztools/x509/json.go (hunk 4 above)
Comment (zakird) [inline ztools/x509/json.go]: validity is part of tbsCertificate and we'd have to change a bunch of code in x509.go

File: ztools/x509/x509.go [hunk 5]
```diff
@@ -740,6 +748,18 @@ func (c *Certificate) CheckSignature(algo SignatureAlgorithm, signed, signature
 			return errors.New("x509: ECDSA verification failure")
 		}
 		return
+	case *AugmentedECDSA:
```
Comment (dadrian) [inline ztools/x509/x509.go]: Will this ever fire? We don't ever deserialize an AugmentedECDSA, do we?

File: ztools/x509/x509.go [hunk 6]
```diff
@@ -859,11 +879,16 @@ func parsePublicKey(algo PublicKeyAlgorithm, keyData *publicKeyInfo) (interface{
...
 			return nil, errors.New("x509: failed to unmarshal elliptic curve point")
 		}
-		pub := &ecdsa.PublicKey{
+		key := &ecdsa.PublicKey{
 			Curve: namedCurve,
 			X:     x,
 			Y:     y,
 		}
+
+		pub := &AugmentedECDSA{
```
Comment (dadrian) [inline ztools/x509/x509.go]: Oh I see, we add it back in here.

File: ztools/x509/json.go (hunk 4 above)
Comment (dadrian) [inline ztools/x509/json.go]: SGTM.

**Comment by dadrian** [comment]
Looks like my only remaining comment is about hours vs seconds.

**Comment by dadrian** [comment]
I think your text editor just replaced a bunch of tabs with spaces O_O

**Comment by zakird** [comment]
@shoegaze this fails tests. 👎

File: ztools/x509/json.go [hunk 7]
```diff
@@ -204,8 +212,9 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.SignatureAlgorithm = c.SignatureAlgorithm
 	jc.Issuer = c.Issuer
 	jc.IssuerDN = c.Issuer.String()
-	jc.Validity.NotBefore = c.NotBefore
+	jc.alidity.NotBefore = c.NotBefore
```
Comment (zzma) [inline ztools/x509/json.go]: typo here

Comment (shoegaze) [inline ztools/x509/json.go]: Well that's awkward.

**Comment by zakird** [comment]
I also made the necessary schema changes to ztag.

**Pull request merged**